modules/stochastic_tools/include/vectorpostprocessors/StochasticResults.h

```diff
@@ -43,6 +43,15 @@ class StochasticResults : public GeneralVectorPostprocessor, SamplerInterface
    */
   VectorPostprocessorValue & getVectorPostprocessorValueByGroup(unsigned int group);
 
+  /**
+   * Get the sample vectors
+   * @return A const pointer to the vector of sample vectors
+   */
+  const std::vector<VectorPostprocessorValue *> * getSampleVectors() const
```

Comment:
Why not use a reference?

Reply:
I could have it return a reference if you prefer, but I'll just end up converting it to a pointer because I can't get at this object until `InitialSetup()`, so my member variable that points to this has to be a pointer (since it can't be set at construction).

Reply:
I would like this to be a ref., I don't like having public API methods that rely on pointers. 

For your case, you need to use a pointer for the StochasticResults object, but then in your code you can just call the method when you need it: `_results_obj_ptr->getSampleVectors()`.

Reply:
OK, I'll make it a reference.